aten/src/ATen/native/native_functions.yaml

```diff
@@ -3267,12 +3267,20 @@
   dispatch:
     CompositeExplicitAutograd: linspace
 
+- func: linspace.Tensor(Tensor start, Tensor end, int steps, *, ScalarType? dtype=None, Layout? layout=None, Device? device=None, bool? pin_memory=None) -> Tensor
```

Comment:
Also, this change is technically BC-breaking. Previously, a model calling `torch.linspace(torch.tensor(0), torch.tensor(10), 10)` would go through the linspace.Scalar overload. Now, it goes through the linspace.Tensor overload.

I hope this doesn't actually matter in practice, but we have a FC/BC guarantee of around two weeks for our aten operators. Our tests (OSS, and then internal) will tell us if everything is OK. If they're not OK, then we're going to need to do the following:
- add linspace.Tensor, but lower its priority below linspace.Scalar so that linspace.Scalar gets triggered.
- wait two weeks
- swap the priority between linspace.Tensor and linspace.Scalar.

TL;DR: it might take some time to actually merge this PR, but we'll get there if we persevere.

Reply:
Edit: It isn't BC-breaking, the right word is not forwards compatible (FC):
- if a producer uses this PR and a consumer uses an older version of PyTorch, then the consumer won't have the new operator. This is an FC problem
- If a producer uses an older version of PyTorch and a consumer uses this PR, we're good, so there are no BC concerns

Given there's not really a good way to test this I think we are just going to yolo merge and see if anyone reverts.